From 7344dcc03bee35161efe83f4fccb5bf1ee79adf6 Mon Sep 17 00:00:00 2001
From: Joakim Plate <elupus@xbmc.org>
Date: Sat, 31 Mar 2012 15:48:15 +0200
Subject: [PATCH 136/145] changed: Only allow slice based parallel decoding

Frame based parallel decoding causes HW decoding
to crash, and it's more sensitive to changes
in frame sizes during decoding.
(cherry picked from commit a0fc63df945440719a724d80ab0afbc28f67263e)
---
 xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp
index 33bb3b7..256bc8b 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp
@@ -239,6 +239,10 @@ bool CDVDVideoCodecFFmpeg::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options
   m_pCodecContext->workaround_bugs = FF_BUG_AUTODETECT;
   m_pCodecContext->get_format = GetFormat;
   m_pCodecContext->codec_tag = hints.codec_tag;
+  /* Only allow slice threading, since frame threading is more
+   * sensitive to changes in frame sizes, and it causes crashes
+   * during HW accell */
+  m_pCodecContext->thread_type = FF_THREAD_SLICE;
 
 #if defined(__APPLE__) && defined(__arm__)
   // ffmpeg with enabled neon will crash and burn if this is enabled
-- 
1.7.10

